From f39e1600c0a3eafa2d248458f429783271e4aa4b Mon Sep 17 00:00:00 2001
From: Patrick Rohr <prohr@google.com>
Date: Thu, 22 Dec 2022 14:18:25 -0800
Subject: [PATCH] cronet: remove local zlib include

Cherry-pick crrev/c/4109079.

Additionally, fixed cronet_library_loader.cc per Motomu's comment.

Test: none
Change-Id: Idba3fe1d06e88105960d0b3c4549dd5a034cb250
---
 components/cronet/android/cronet_library_loader.cc              | 2 +-
 net/disk_cache/simple/simple_entry_impl.cc                      | 2 +-
 net/disk_cache/simple/simple_synchronous_entry.cc               | 2 +-
 net/disk_cache/simple/simple_util.cc                            | 2 +-
 net/disk_cache/simple/simple_version_upgrade.cc                 | 2 +-
 net/filter/filter_source_stream_test_util.cc                    | 2 +-
 net/filter/gzip_header.cc                                       | 2 +-
 net/filter/gzip_source_stream.cc                                | 2 +-
 net/filter/gzip_source_stream_unittest.cc                       | 2 +-
 net/tools/content_decoder_tool/content_decoder_tool_unittest.cc | 2 +-
 net/websockets/websocket_deflater.cc                            | 2 +-
 net/websockets/websocket_inflater.cc                            | 2 +-
 12 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/components/cronet/android/cronet_library_loader.cc b/components/cronet/android/cronet_library_loader.cc
index f25be3bfe8146..ec5c19bc42b47 100644
--- a/components/cronet/android/cronet_library_loader.cc
+++ b/components/cronet/android/cronet_library_loader.cc
@@ -6,6 +6,7 @@
 #include <memory>
 #include <string>
 #include <utility>
+#include <zlib.h>
 
 #include "base/android/base_jni_onload.h"
 #include "base/android/build_info.h"
@@ -30,7 +31,6 @@
 #include "net/base/network_change_notifier.h"
 #include "net/proxy_resolution/configured_proxy_resolution_service.h"
 #include "net/proxy_resolution/proxy_config_service_android.h"
-#include "third_party/zlib/zlib.h"
 #include "url/buildflags.h"
 
 #if !BUILDFLAG(USE_PLATFORM_ICU_ALTERNATIVES)
diff --git a/net/disk_cache/simple/simple_entry_impl.cc b/net/disk_cache/simple/simple_entry_impl.cc
index 4bfa63b37f23d..f84eab0ecf3f4 100644
--- a/net/disk_cache/simple/simple_entry_impl.cc
+++ b/net/disk_cache/simple/simple_entry_impl.cc
@@ -9,6 +9,7 @@
 #include <limits>
 #include <utility>
 #include <vector>
+#include <zlib.h>
 
 #include "base/bind.h"
 #include "base/callback.h"
@@ -36,7 +37,6 @@
 #include "net/disk_cache/simple/simple_util.h"
 #include "net/log/net_log.h"
 #include "net/log/net_log_source_type.h"
-#include "third_party/zlib/zlib.h"
 
 namespace disk_cache {
 namespace {
diff --git a/net/disk_cache/simple/simple_synchronous_entry.cc b/net/disk_cache/simple/simple_synchronous_entry.cc
index cbeaf88f1fb2d..933ce29dfcfe9 100644
--- a/net/disk_cache/simple/simple_synchronous_entry.cc
+++ b/net/disk_cache/simple/simple_synchronous_entry.cc
@@ -8,6 +8,7 @@
 #include <cstring>
 #include <functional>
 #include <limits>
+#include <zlib.h>
 
 #include "base/compiler_specific.h"
 #include "base/containers/stack_container.h"
@@ -33,7 +34,6 @@
 #include "net/disk_cache/simple/simple_histogram_enums.h"
 #include "net/disk_cache/simple/simple_histogram_macros.h"
 #include "net/disk_cache/simple/simple_util.h"
-#include "third_party/zlib/zlib.h"
 
 using base::FilePath;
 using base::Time;
diff --git a/net/disk_cache/simple/simple_util.cc b/net/disk_cache/simple/simple_util.cc
index b29ff3fb1af3c..9a466cef8d750 100644
--- a/net/disk_cache/simple/simple_util.cc
+++ b/net/disk_cache/simple/simple_util.cc
@@ -5,6 +5,7 @@
 #include "net/disk_cache/simple/simple_util.h"
 
 #include <limits>
+#include <zlib.h>
 
 #include "base/check_op.h"
 #include "base/files/file_util.h"
@@ -16,7 +17,6 @@
 #include "base/threading/thread_restrictions.h"
 #include "base/time/time.h"
 #include "net/disk_cache/simple/simple_entry_format.h"
-#include "third_party/zlib/zlib.h"
 
 namespace {
 
diff --git a/net/disk_cache/simple/simple_version_upgrade.cc b/net/disk_cache/simple/simple_version_upgrade.cc
index 09d80c5af8722..1d6bcce1184fe 100644
--- a/net/disk_cache/simple/simple_version_upgrade.cc
+++ b/net/disk_cache/simple/simple_version_upgrade.cc
@@ -5,6 +5,7 @@
 #include "net/disk_cache/simple/simple_version_upgrade.h"
 
 #include <cstring>
+#include <zlib.h>
 
 #include "base/files/file.h"
 #include "base/files/file_enumerator.h"
@@ -16,7 +17,6 @@
 #include "net/disk_cache/disk_cache.h"
 #include "net/disk_cache/simple/simple_backend_version.h"
 #include "net/disk_cache/simple/simple_entry_format_history.h"
-#include "third_party/zlib/zlib.h"
 
 namespace {
 
diff --git a/net/filter/filter_source_stream_test_util.cc b/net/filter/filter_source_stream_test_util.cc
index 2e18235b72607..73ccfb75e01ef 100644
--- a/net/filter/filter_source_stream_test_util.cc
+++ b/net/filter/filter_source_stream_test_util.cc
@@ -5,10 +5,10 @@
 #include "net/filter/filter_source_stream_test_util.h"
 
 #include <cstring>
+#include <zlib.h>
 
 #include "base/bit_cast.h"
 #include "base/check_op.h"
-#include "third_party/zlib/zlib.h"
 
 namespace net {
 
diff --git a/net/filter/gzip_header.cc b/net/filter/gzip_header.cc
index 69028c17b4bde..b0b1ed62bb6ff 100644
--- a/net/filter/gzip_header.cc
+++ b/net/filter/gzip_header.cc
@@ -7,9 +7,9 @@
 #include <string.h>
 
 #include <algorithm>
+#include <zlib.h>
 
 #include "base/check_op.h"
-#include "third_party/zlib/zlib.h"
 
 namespace net {
 
diff --git a/net/filter/gzip_source_stream.cc b/net/filter/gzip_source_stream.cc
index 129ca795692d6..c51f4b97c89ab 100644
--- a/net/filter/gzip_source_stream.cc
+++ b/net/filter/gzip_source_stream.cc
@@ -7,6 +7,7 @@
 #include <algorithm>
 #include <memory>
 #include <utility>
+#include <zlib.h>
 
 #include "base/bind.h"
 #include "base/bit_cast.h"
@@ -16,7 +17,6 @@
 #include "base/notreached.h"
 #include "base/numerics/checked_math.h"
 #include "net/base/io_buffer.h"
-#include "third_party/zlib/zlib.h"
 
 namespace net {
 
diff --git a/net/filter/gzip_source_stream_unittest.cc b/net/filter/gzip_source_stream_unittest.cc
index c4a109c69d7ed..cb018f4924777 100644
--- a/net/filter/gzip_source_stream_unittest.cc
+++ b/net/filter/gzip_source_stream_unittest.cc
@@ -4,6 +4,7 @@
 
 #include <string>
 #include <utility>
+#include <zlib.h>
 
 #include "base/bind.h"
 #include "base/callback.h"
@@ -14,7 +15,6 @@
 #include "net/filter/gzip_source_stream.h"
 #include "net/filter/mock_source_stream.h"
 #include "testing/gtest/include/gtest/gtest.h"
-#include "third_party/zlib/zlib.h"
 
 namespace net {
 
diff --git a/net/tools/content_decoder_tool/content_decoder_tool_unittest.cc b/net/tools/content_decoder_tool/content_decoder_tool_unittest.cc
index c278138944463..07c9059d56993 100644
--- a/net/tools/content_decoder_tool/content_decoder_tool_unittest.cc
+++ b/net/tools/content_decoder_tool/content_decoder_tool_unittest.cc
@@ -8,6 +8,7 @@
 #include <memory>
 #include <ostream>
 #include <utility>
+#include <zlib.h>
 
 #include "base/files/file_util.h"
 #include "base/path_service.h"
@@ -16,7 +17,6 @@
 #include "net/filter/mock_source_stream.h"
 #include "testing/gtest/include/gtest/gtest.h"
 #include "testing/platform_test.h"
-#include "third_party/zlib/zlib.h"
 
 namespace net {
 
diff --git a/net/websockets/websocket_deflater.cc b/net/websockets/websocket_deflater.cc
index 0a66a66249129..ec9a257906171 100644
--- a/net/websockets/websocket_deflater.cc
+++ b/net/websockets/websocket_deflater.cc
@@ -7,11 +7,11 @@
 #include <string.h>
 #include <algorithm>
 #include <vector>
+#include <zlib.h>
 
 #include "base/check_op.h"
 #include "base/containers/circular_deque.h"
 #include "net/base/io_buffer.h"
-#include "third_party/zlib/zlib.h"
 
 namespace net {
 
diff --git a/net/websockets/websocket_inflater.cc b/net/websockets/websocket_inflater.cc
index ac4f6934a99fc..00ede96566169 100644
--- a/net/websockets/websocket_inflater.cc
+++ b/net/websockets/websocket_inflater.cc
@@ -6,10 +6,10 @@
 
 #include <algorithm>
 #include <vector>
+#include <zlib.h>
 
 #include "base/check_op.h"
 #include "net/base/io_buffer.h"
-#include "third_party/zlib/zlib.h"
 
 namespace net {
 
-- 
2.39.0.314.g84b9a713c41-goog

